moped-editor/src/views/projects/projectView/ProjectComponents/utils/map.js

```diff
@@ -126,32 +126,31 @@ export const zoomMapToFeatureCollection = (
 /**
  * Use Mapbox fitBounds to zoom to existing project components feature collection
  * @param {Object} mapRef - React ref that stores the Mapbox map instance (mapRef.current)
- * @param {Object} data - Data returned from the moped_components query
- * @param {Array} data.project_geography - Array of existing component features
+ * @param {Object} featureCollection - feature collection of project components
+ * @param {boolean} refreshOnComponentsUpdate - if true, zoom to components on every update
  */
-export const useZoomToExistingComponents = (mapRef, data) => {
+export const useZoomToExistingComponents = (
+  mapRef,
+  featureCollection,
+  refreshOnComponentsUpdate = false
+) => {
   const [hasMapZoomedInitially, setHasMapZoomedInitially] = useState(false);
 
   useEffect(() => {
-    if (!data || hasMapZoomedInitially) return;
+    if (!featureCollection || hasMapZoomedInitially) return;
     if (!mapRef?.current) return;
 
-    if (data.project_geography.length === 0) {
-      setHasMapZoomedInitially(true);
-      return;
-    }
-
-    const featureCollection = {
```

Comment:
We don't need to build a feature collection since this hook now takes one as an argument.